[2025-10-05T09:23:59.565+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-10-05T09:23:59.579+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T09:15:00+00:00 [queued]>
[2025-10-05T09:23:59.587+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T09:15:00+00:00 [queued]>
[2025-10-05T09:23:59.587+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 3
[2025-10-05T09:23:59.600+0000] {taskinstance.py:2888} INFO - Executing <Task(_PythonDecoratedOperator): train_and_log_models> on 2025-10-05 09:15:00+00:00
[2025-10-05T09:23:59.608+0000] {standard_task_runner.py:72} INFO - Started process 6775 to run task
[2025-10-05T09:23:59.610+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'forest_cover_mlops_pipeline', 'train_and_log_models', 'scheduled__2025-10-05T09:15:00+00:00', '--job-id', '112', '--raw', '--subdir', 'DAGS_FOLDER/forest_cover_dag.py', '--cfg-path', '/home/airflow/tmp/tmp_63jrxwh']
[2025-10-05T09:23:59.611+0000] {standard_task_runner.py:105} INFO - Job 112: Subtask train_and_log_models
[2025-10-05T09:23:59.660+0000] {task_command.py:467} INFO - Running <TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T09:15:00+00:00 [running]> on host 3702e9b7dcde
[2025-10-05T09:24:00.198+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='mlops-team' AIRFLOW_CTX_DAG_ID='forest_cover_mlops_pipeline' AIRFLOW_CTX_TASK_ID='train_and_log_models' AIRFLOW_CTX_EXECUTION_DATE='2025-10-05T09:15:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-05T09:15:00+00:00'
[2025-10-05T09:24:00.198+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-10-05T09:24:00.359+0000] {logging_mixin.py:190} INFO - Entrenando 3 modelos con GridSearchCV...
[2025-10-05T09:24:00.359+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T09:24:00.359+0000] {logging_mixin.py:190} INFO - Modelo: RandomForest
[2025-10-05T09:24:00.359+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T09:24:00.375+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:24:00 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet
[2025-10-05T09:24:00.427+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T09:24:00.428+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 12 candidates, totalling 36 fits
[2025-10-05T09:25:19.947+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__max_depth': None, 'classifier__min_samples_split': 2, 'classifier__n_estimators': 100}
[2025-10-05T09:25:30.028+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 1.0000
[2025-10-05T09:25:30.028+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.9814
[2025-10-05T09:25:30.028+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.9814
[2025-10-05T09:25:30.028+0000] {logging_mixin.py:190} INFO -   CV Score: 0.9734 (+/- 0.0015)
[2025-10-05T09:25:32.180+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:25:32 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T09:25:32.230+0000] {credentials.py:1147} INFO - Found credentials in environment variables.
[2025-10-05T09:25:32.511+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_randomforest' already exists. Creating a new version of this model...
[2025-10-05T09:25:32.527+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:25:32 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_randomforest, version 11
[2025-10-05T09:25:32.527+0000] {logging_mixin.py:190} WARNING - Created version '11' of model 'forest_cover_randomforest'.
[2025-10-05T09:25:32.537+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:25:32 INFO mlflow.tracking._tracking_service.client: üèÉ View run RandomForest_batch_10 at: http://mlflow:5000/#/experiments/1/runs/2cb4319d1119462ea0470a50e7c225b4.
[2025-10-05T09:25:32.538+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:25:32 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T09:25:32.549+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T09:25:32.549+0000] {logging_mixin.py:190} INFO - Modelo: GradientBoosting
[2025-10-05T09:25:32.550+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T09:25:32.601+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T09:25:32.601+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 8 candidates, totalling 24 fits
[2025-10-05T09:36:40.558+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__learning_rate': 0.1, 'classifier__max_depth': 5, 'classifier__n_estimators': 100}
[2025-10-05T09:38:45.533+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 0.9442
[2025-10-05T09:38:45.533+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.9293
[2025-10-05T09:38:45.534+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.9397
[2025-10-05T09:38:45.534+0000] {logging_mixin.py:190} INFO -   CV Score: 0.9497 (+/- 0.0013)
[2025-10-05T09:38:47.037+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:38:47 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T09:38:47.164+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_gradientboosting' already exists. Creating a new version of this model...
[2025-10-05T09:38:47.210+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:38:47 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_gradientboosting, version 11
[2025-10-05T09:38:47.210+0000] {logging_mixin.py:190} WARNING - Created version '11' of model 'forest_cover_gradientboosting'.
[2025-10-05T09:38:47.238+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:38:47 INFO mlflow.tracking._tracking_service.client: üèÉ View run GradientBoosting_batch_10 at: http://mlflow:5000/#/experiments/1/runs/53cc882513cd4163905afad6ad2d2da0.
[2025-10-05T09:38:47.238+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:38:47 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T09:38:47.258+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T09:38:47.258+0000] {logging_mixin.py:190} INFO - Modelo: LogisticRegression
[2025-10-05T09:38:47.259+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T09:38:47.375+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T09:38:47.376+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 6 candidates, totalling 18 fits
[2025-10-05T09:39:43.383+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T09:39:52.176+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T09:40:01.102+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T09:40:07.961+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__C': 10.0, 'classifier__solver': 'lbfgs'}
[2025-10-05T09:40:26.315+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 0.7629
[2025-10-05T09:40:26.316+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.7654
[2025-10-05T09:40:26.318+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.7554
[2025-10-05T09:40:26.319+0000] {logging_mixin.py:190} INFO -   CV Score: 0.7623 (+/- 0.0023)
[2025-10-05T09:40:27.927+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:40:27 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T09:40:28.014+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_logisticregression' already exists. Creating a new version of this model...
[2025-10-05T09:40:28.069+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:40:28 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_logisticregression, version 11
[2025-10-05T09:40:28.069+0000] {logging_mixin.py:190} WARNING - Created version '11' of model 'forest_cover_logisticregression'.
[2025-10-05T09:40:28.091+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:40:28 INFO mlflow.tracking._tracking_service.client: üèÉ View run LogisticRegression_batch_10 at: http://mlflow:5000/#/experiments/1/runs/cff804cb08f244fe86bcc3ce7039200e.
[2025-10-05T09:40:28.091+0000] {logging_mixin.py:190} WARNING - 2025/10/05 09:40:28 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T09:40:28.106+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T09:40:28.107+0000] {logging_mixin.py:190} INFO - ‚úì Entrenamiento completado
[2025-10-05T09:40:28.107+0000] {logging_mixin.py:190} INFO -   Mejor modelo: RandomForest (Accuracy: 0.9814)
[2025-10-05T09:40:28.108+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T09:40:28.108+0000] {python.py:240} INFO - Done. Returned value was: {'results': {'RandomForest': {'test_accuracy': 0.9813800657174151, 'test_f1_weighted': 0.9813509338815236, 'cv_mean_score': 0.9733805324292898, 'best_params': {'classifier__max_depth': None, 'classifier__min_samples_split': 2, 'classifier__n_estimators': 100}}, 'GradientBoosting': {'test_accuracy': 0.9292755437333751, 'test_f1_weighted': 0.9397225890121678, 'cv_mean_score': 0.9497144065684785, 'best_params': {'classifier__learning_rate': 0.1, 'classifier__max_depth': 5, 'classifier__n_estimators': 100}}, 'LogisticRegression': {'test_accuracy': 0.7653731810358316, 'test_f1_weighted': 0.7553973687624423, 'cv_mean_score': 0.7622829599992227, 'best_params': {'classifier__C': 10.0, 'classifier__solver': 'lbfgs'}}}, 'best_model': 'RandomForest', 'best_score': 0.9813800657174151, 'experiment_name': 'forest_cover_classification', 'mlflow_uri': 'http://mlflow:5000'}
[2025-10-05T09:40:28.186+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-10-05T09:40:28.187+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=forest_cover_mlops_pipeline, task_id=train_and_log_models, run_id=scheduled__2025-10-05T09:15:00+00:00, execution_date=20251005T091500, start_date=20251005T092359, end_date=20251005T094028
[2025-10-05T09:40:28.251+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-10-05T09:40:28.267+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-10-05T09:40:28.269+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
